dear_petition/petition/etl/email.py

```diff
@@ -0,0 +1,47 @@
+import logging
+
+from dear_petition.users.models import User
+from dear_petition.petition.etl.load import import_ciprs_records
+
+from dear_petition.sendgrid.models import Email
+
+
+logger = logging.getLogger(__name__)
+
+
+def extract_username_and_label(addr):
+    """
+    Extracts email into parts for importing.
+
+    For example:
+        first.last+mylabel@example.com -> first.last, mylabel
+    """
+    email_username = addr.split("@")[0]
+    username_parts = email_username.split("+")
+    username = username_parts[0]
+    try:
+        label = username_parts[1]
+    except IndexError:
+        label = ""
+    return username, label
+
+
+def sendgrid_webhook_listener(email_id):
+    """
+    Associate email attachments to a user and pass off to the ETL process.
+    """
+    email = Email.objects.get(id=email_id)
+    username, label = extract_username_and_label(email.recipient)
+    try:
+        user = User.objects.get(username=username)
+    except User.DoesNotExist:
+        logger.error(f"Username {username} not found for email '{email}'")
```

Comment:
Just thinking.. is there anything else we should do in the case where we can't extract a username? Should we have a bin of unmatched emails that we clean out periodically? Would we be able to match emails that should have been mapped to a user/batch but weren't?

Reply:
`Batch.user` is currently required, so it's not possible using the current model, but the `Email` and `Attachment` objects will still exist even if a matching user isn't found. Logger errors are picked up by Sentry, so we will be notified. I think it's a good idea though that we can ask them about.

Reply:
Did you ask about this in the meeting? Curious if it came up